### ISSUE TITLE
Understand why the early stop is not really working

### ISSUE THREAD

**Issue by mauricioaniche** [refactoring-ai/Machine-Learning#7]
I have the impression that the early stop mechanism is not really working. We need to double check as it would save lots of processing time.

**Comment by mauricioaniche** [comment]
Some insights from the log: 
`/home/maniche/.local/lib/python3.6/site-packages/keras/callbacks.py:569: RuntimeWarning: Early stopping conditioned on metric `val_loss` which is not available. Available metrics are: loss,acc,precision,recall`
